blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,261 @@
+        public static class DescriptorImpl extends AbstractFolderPropertyDescriptor {
+        }
+
+        public static class CredentialsStoreActionImpl extends CredentialsStoreAction {
+
+            private final CredentialsStore store;
+
+            CredentialsStoreActionImpl(CredentialsStore store) {
+                this.store = store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Nonnull
+            @Override
+            public CredentialsStore getStore() {
+                return store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconFileName() {
+                return isVisible()
+                        ? "/plugin/credentials/images/48x48/folder-store.png"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconClassName() {
+                return isVisible()
+                        ? "icon-credentials-folder-store"
+                        : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getDisplayName() {
+                return Messages.BlueOceanCredentialsProvider_DisplayName();
+            }
+
+            @Override
+            public String getUrlName() {
+                return super.getUrlName();
+            }
+        }
+
+        public static class StoreImpl extends CredentialsStore{
+            private final AbstractFolder abstractFolder;
+            private final CredentialsStoreAction storeAction;
+
+            StoreImpl(AbstractFolder abstractFolder) {
+                super(BlueOceanCredentialsProvider.class);
+                this.abstractFolder = abstractFolder;
+                this.storeAction = new CredentialsStoreActionImpl(this);
+            }
+
+            @Nonnull
+            @Override
+            public List<Domain> getDomains() {
+                //XXX: how to get domain from given Credentials object? Maybe attach domain name to AbstractFolder?
+                return Collections.emptyList();
+            }
+
+            @Nullable
+            @Override
+            public CredentialsStoreAction getStoreAction() {
+                return storeAction;
+            }
+
+            @Nonnull
+            @Override
+            public ModelObject getContext() {
+                return abstractFolder;
+            }
+
+            @Override
+            public boolean hasPermission(@Nonnull Authentication a, @Nonnull Permission permission) {
+                return abstractFolder.getACL().hasPermission(a,permission);
+            }
+
+            @Nonnull
+            @Override
+            public List<Credentials> getCredentials(@Nonnull Domain domain) {
+                return BlueOceanCredentialsProvider.getCredentials(Credentials.class, abstractFolder);
```

Comment:
@stephenc  blueocean allowes credentials to be stored in different domain. For example GitHub credentials could be stored in different domain than git credentials.